src/high-level/src/KinDynComputations.cpp

```diff
@@ -1073,6 +1073,27 @@ Transform KinDynComputations::getWorldTransform(const FrameIndex frameIndex)
     return world_H_frame;
 }
 
+std::vector<iDynTree::Matrix4x4> KinDynComputations::getWorldTransformsAsHomogeneous(const std::vector<std::string>& frameNames)
+{
+    std::vector<iDynTree::Matrix4x4> worldTransforms;
+    int numberOfTransforms=frameNames.size();
+    for(int number=0; number<numberOfTransforms; number++)
+    {
+        std::string frameName=frameNames[number];
+        int frameIndex = getFrameIndex(frameName);
+        if( frameIndex < 0 )
```

Comment:
The correct check is frameIndex == iDynTree::FRAME_INVALID_INDEX:
```suggestion
        if (frameIndex  == iDynTree::FRAME_INVALID_INDEX)
```

Reply:
Will do